File: lib/models/ticket.dart
class Ticket {
  final String id;
  final String eventId;
  final String userId;
  final String type;
  final double price;
  final int quantity;
  final String status;
  final String? qrCode;
  final DateTime createdAt;

  Ticket({
    required this.id,
    required this.eventId,
    required this.userId,
    required this.type,
    required this.price,
    required this.quantity,
    required this.status,
    this.qrCode,
    required this.createdAt,
  });

  factory Ticket.fromJson(Map<String, dynamic> json) {
    return Ticket(
      id: json['_id'],
      eventId: json['event'],
      userId: json['user'],
      type: json['type'],
      price: json['price'].toDouble(),
      quantity: json['quantity'],
      status: json['status'],
      qrCode: json['qrCode'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }
} 

File: nextmit-workspace/nextmit-backend/src/models/Ticket.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface ITicket extends Document {
  event: mongoose.Types.ObjectId;
  user: mongoose.Types.ObjectId;
  type: string;
  price: number;
  quantity: number;
  status: 'pending' | 'confirmed' | 'cancelled' | 'used';
  paymentId?: string;
  qrCode?: string;
  createdAt: Date;
  updatedAt: Date;
}

const ticketSchema = new Schema<ITicket>({
  event: {
    type: Schema.Types.ObjectId,
    ref: 'Event',
    required: true,
  },
  user: {
    type: Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  type: {
    type: String,
    required: [true, 'Le type de ticket est requis'],
  },
  price: {
    type: Number,
    required: [true, 'Le prix est requis'],
    min: [0, 'Le prix ne peut pas être négatif'],
  },
  quantity: {
    type: Number,
    required: [true, 'La quantité est requise'],
    min: [1, 'La quantité doit être d\'au moins 1'],
  },
  status: {
    type: String,
    enum: ['pending', 'confirmed', 'cancelled', 'used'],
    default: 'pending',
  },
  paymentId: String,
  qrCode: String,
}, {
  timestamps: true,
});

// Index pour la recherche rapide
ticketSchema.index({ event: 1, user: 1 });
ticketSchema.index({ status: 1 });

// Méthodes virtuelles
ticketSchema.virtual('totalPrice').get(function() {
  return this.price * this.quantity;
});

export const Ticket = mongoose.model<ITicket>('Ticket', ticketSchema); 